gpflow/optimizers/scipy.py

```diff
@@ -34,44 +34,65 @@ def minimize(self,
         if not callable(closure):
             raise TypeError('Callable object expected.')  # pragma: no cover
         initial_params = self.initial_parameters(variables)
-        func = self.eval_func(closure, variables, step_callback)
+        func = self.eval_func(closure, variables)
+        if step_callback is not None:
+            if 'callback' in scipy_kwargs:
+                raise ValueError("Callback passed both via `step_callback` and `callback`")
+
+            callback = self.callback_func(closure, variables, step_callback)
```

Comment:
If the default value is None, it should stay None. Otherwise, it is just calling of the empty callback wasting cpu clock time (python function call isn't like c function call :)

Reply:
I don't understand what you mean - that's why I have the `if step_callback is not None:` check?

Reply:
Or do you want to replace the `if 'callback' in scipy_kwargs:` with `if scipy_kwargs.get('callback', None) is not None:`?